src/test/java/com/jcabi/github/RtPublicMembersTest.java

```diff
@@ -0,0 +1,259 @@
+import com.jcabi.http.mock.MkContainer;
+import com.jcabi.http.mock.MkGrizzlyContainer;
+import com.jcabi.http.mock.MkQuery;
+import com.jcabi.http.request.ApacheRequest;
+import com.jcabi.http.request.FakeRequest;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/**
+ * Test case for {@link RtPublicMembers}.
+ *
+ * @author Chris Rebert (github@chrisrebert.com)
+ * @version $Id$
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+public final class RtPublicMembersTest {
+    /**
+     * Test organization.
+     */
+    private static final String ORG = "starfleet";
+
+    /**
+     * Test username.
+     */
+    private static final String USERNAME = "wesley";
+
+    /**
+     * Public members URL for test org.
+     */
+    private static final String MEMBERS_URL = String.format(
+        "/orgs/%s/public_members",
+        ORG
+    );
+
+    /**
+     * Public member URL for test user in test org.
+     */
+    private static final String MEMBER_URL = String.format(
+        "%s/%s",
+        MEMBERS_URL,
+        USERNAME
+    );
+
+    /**
+     * Rule for checking thrown exception.
+     * @checkstyle VisibilityModifier (3 lines)
+     */
+    @Rule
+    public transient ExpectedException thrown = ExpectedException.none();
+
+    /**
+     * RtPublicMembers can fetch its organization.
+     * @throws IOException If there is an I/O problem
+     */
+    @Test
+    public void fetchesOrg() throws IOException {
+        final Organization org = organization();
+        MatcherAssert.assertThat(
+            new RtPublicMembers(new FakeRequest(), org).org(),
+            Matchers.equalTo(org)
+        );
+    }
+
+    /**
+     * RtPublicMembers can conceal a user's membership in the organization.
+     * @throws IOException If there is an I/O problem
+     */
+    @Test
+    public void concealsMembers() throws IOException {
+        final MkContainer container = new MkGrizzlyContainer()
+            .next(new MkAnswer.Simple(HttpURLConnection.HTTP_NO_CONTENT))
+            .next(new MkAnswer.Simple(HttpURLConnection.HTTP_INTERNAL_ERROR))
+            .start();
+        final RtPublicMembers members = new RtPublicMembers(
+            new ApacheRequest(container.home()),
+            organization()
+        );
+        members.conceal(user());
+        final MkQuery req = container.take();
+        MatcherAssert.assertThat(
+            req.method(),
+            Matchers.equalTo(Request.DELETE)
+        );
+        MatcherAssert.assertThat(
+            req.body(),
+            Matchers.isEmptyOrNullString()
+        );
+        MatcherAssert.assertThat(
+            req.uri().toString(),
+            Matchers.endsWith(MEMBER_URL)
+        );
+        this.thrown.expect(AssertionError.class);
+        members.conceal(user());
```

Comment:
@cvrebert I've got it ... basically, we have several logical test cases in one test method ... that's why we use this weird exception interceptor instead of just declaring expected exception in the ```@Test``` annotation